internal/ledger/compare_test.go

```diff
@@ -0,0 +1,597 @@
+			blockNum: 1, txNum: 2, metadata: "md3",
+		},
+		{
+			namespace: "ns3", key: "k1", value: "v4",
+			blockNum: 2, txNum: 1, metadata: "md4",
+		},
+	}
+
+	var sampleRecords3 []*testRecord = []*testRecord{
+		{
+			namespace: "ns1", key: "k1", value: "v1",
+			blockNum: 1, txNum: 1, metadata: "md1",
+		},
+		{
+			namespace: "ns2", key: "k1", value: "v3",
+			blockNum: 1, txNum: 2, metadata: "md3",
+		},
+		{
+			namespace: "ns3", key: "k1", value: "v4",
+			blockNum: 2, txNum: 1, metadata: "md4",
+		},
+	}
+
+	var sampleRecords4 []*testRecord = []*testRecord{
+		{
+			namespace: "ns1", key: "k1", value: "v1",
+			blockNum: 1, txNum: 1, metadata: "md1",
+		},
+		{
+			namespace: "ns1", key: "k2", value: "v2",
+			blockNum: 1, txNum: 1, metadata: "md2",
+		},
+	}
+
+	// Signable metadata samples for snapshots
+	sampleSignableMetadata1 := &kvledger.SnapshotSignableMetadata{
+		ChannelName:            "testchannel",
+		LastBlockNumber:        sampleRecords1[len(sampleRecords1)-1].blockNum,
+		LastBlockHashInHex:     "last_block_hash",
+		PreviousBlockHashInHex: "previous_block_hash",
+		FilesAndHashes: map[string]string{
+			"private_state_hashes.data":     "private_state_hash1",
+			"private_state_hashes.metadata": "private_state_hash1",
+			"public_state.data":             "public_state_hash1",
+			"public_state.metadata":         "public_state_hash1",
+			"txids.data":                    "txids_hash1",
+			"txids.metadata":                "txids_hash1",
+		},
+		StateDBType: "testdatabase",
+	}
+
+	sampleSignableMetadata2 := &kvledger.SnapshotSignableMetadata{
+		ChannelName:            "testchannel",
+		LastBlockNumber:        sampleRecords1[len(sampleRecords1)-1].blockNum,
+		LastBlockHashInHex:     "last_block_hash",
+		PreviousBlockHashInHex: "previous_block_hash",
+		FilesAndHashes: map[string]string{
+			"private_state_hashes.data":     "private_state_hash2",
+			"private_state_hashes.metadata": "private_state_hash2",
+			"public_state.data":             "public_state_hash2",
+			"public_state.metadata":         "public_state_hash2",
+			"txids.data":                    "txids_hash2",
+			"txids.metadata":                "txids_hash2",
+		},
+		StateDBType: "testdatabase",
+	}
+
+	sampleSignableMetadata3 := &kvledger.SnapshotSignableMetadata{
+		ChannelName:            "testchannel",
+		LastBlockNumber:        sampleRecords1[len(sampleRecords1)-1].blockNum,
+		LastBlockHashInHex:     "last_block_hash",
+		PreviousBlockHashInHex: "previous_block_hash",
+		FilesAndHashes: map[string]string{
+			"private_state_hashes.data":     "private_state_hash2",
+			"private_state_hashes.metadata": "private_state_hash2",
+			"public_state.data":             "public_state_hash2",
+			"public_state.metadata":         "public_state_hash2",
+			"txids.data":                    "txids_hash2",
+			"txids.metadata":                "txids_hash2",
+		},
+		StateDBType: "testdatabase2",
+	}
+
+	// Expected outputs
+	const (
+		expectedDifferenceResult = `[
```

Comment:
just like other variables above
```suggestion
		expectedDifferenceResult := `[
```